From 64f640a14e48171af9322012d8948bcc5d15c036 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Fr=C3=A9d=C3=A9ric=20Pierret=20=28fepitre=29?=
 <frederic.pierret@qubes-os.org>
Date: Sun, 15 Dec 2019 16:47:44 +0100
Subject: [PATCH] Mark 'qubes' user name as reserved

'qubes' group is used internally, but useradd want to create a new group
named as new user, so 'qubes' user name should also be avoided.

Fixes QubesOS/qubes-issues#3777

From 7a60d316ea32df7f9a0c25669b20a0b608a2e384
By Marek Marczykowski <marmarek@invisiblethingslab.com>
---
 pyanaconda/core/users.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pyanaconda/core/users.py b/pyanaconda/core/users.py
index 9de29b24a1..bcfe2da51b 100644
--- a/pyanaconda/core/users.py
+++ b/pyanaconda/core/users.py
@@ -63,7 +63,7 @@ def check_username(name):
     """
 
     # Check reserved names.
-    if name in os.listdir("/") + ["root", "home", "daemon", "system"]:
+    if name in os.listdir("/") + ["root", "home", "daemon", "system", "qubes"]:
         return False, _("User name is reserved for system: %s") % name
 
     return is_valid_name(name)
-- 
2.31.1

